routes/index.js

```diff
@@ -35,7 +35,20 @@ exports.index = function (req, res, next) {
 };
 
 exports.loginHandler = function (req, res, next) {
-  throw new Error("Not implemented")
+  if (validator.isEmail(req.body.username)) {
+    User.find({ username: req.body.username, password: req.body.password }, function (err, users) {
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg" width="20"/></sub>&nbsp; NoSQL Injection

Unsanitized input from the HTTP request body flows into find, where it is used in an NoSQL query. This may result in an NoSQL Injection vulnerability.

<h6>
<a href="https://cwe.mitre.org/data/definitions/943.html">CWE-943</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 814</a> | Line 39</h6>

<details>

<summary><b>Data flow: 6 steps</b></summary>

`Step 1` https://github.com/trailblazing-dx/tiago_app.dev.snyk_github_oauth_nodejs-goof/blob/5474b30db9f4aebf76dcd7432508f467ed6e81f8/routes/index.js#39

`Step 2` [routes/index.js#39](https://github.com/trailblazing-dx/tiago_app.dev.snyk_github_oauth_nodejs-goof/blob/5474b30db9f4aebf76dcd7432508f467ed6e81f8/routes/index.js#39)

`Step 3` [routes/index.js#39](https://github.com/trailblazing-dx/tiago_app.dev.snyk_github_oauth_nodejs-goof/blob/5474b30db9f4aebf76dcd7432508f467ed6e81f8/routes/index.js#39)

`Step 4` [routes/index.js#39](https://github.com/trailblazing-dx/tiago_app.dev.snyk_github_oauth_nodejs-goof/blob/5474b30db9f4aebf76dcd7432508f467ed6e81f8/routes/index.js#39)

`Step 5` [routes/index.js#39](https://github.com/trailblazing-dx/tiago_app.dev.snyk_github_oauth_nodejs-goof/blob/5474b30db9f4aebf76dcd7432508f467ed6e81f8/routes/index.js#39)

`Step 6` https://github.com/trailblazing-dx/tiago_app.dev.snyk_github_oauth_nodejs-goof/blob/5474b30db9f4aebf76dcd7432508f467ed6e81f8/routes/index.js#39

</details>